src/server/game/Entities/Player/Player.cpp

```diff
@@ -20236,33 +20237,47 @@ void Player::SendOnCancelExpectedVehicleRideAura()
 
 void Player::PetSpellInitialize()
 {
-    Pet* pet = GetPet();
+	Pet* pet = GetPet();
 
-    if (!pet)
-        return;
+	if (!pet)
+		return;
 
-    TC_LOG_DEBUG("entities.pet", "Pet Spells Groups");
+	TC_LOG_DEBUG("entities.pet", "Pet Spells Groups");
 
-    CharmInfo* charmInfo = pet->GetCharmInfo();
+	CharmInfo* charmInfo = pet->GetCharmInfo();
 
-    WorldPacket data(SMSG_PET_SPELLS_MESSAGE, 8 + 2 + 4 + 4 + 4 * MAX_UNIT_ACTION_BAR_INDEX + 1 + 1);
-    data << pet->GetGUID();
+	WorldPacket data(SMSG_PET_SPELLS_MESSAGE, 16 + 2 + 2 + 4 + 1 + 1 + 2 + 4 * MAX_UNIT_ACTION_BAR_INDEX + 4 + 4 + 4);
```

Comment:
Needs to be converted to packet classes. Any packet writing or reading outside of Read() and Write() packet class methods is deprecated.